app/src/main/java/fr/iut_rodez/pathpilot_android_client/MainActivity.java

```diff
@@ -1,24 +1,199 @@
 package fr.iut_rodez.pathpilot_android_client;
 
 import android.os.Bundle;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
 
 import androidx.activity.EdgeToEdge;
 import androidx.appcompat.app.AppCompatActivity;
-import androidx.core.graphics.Insets;
-import androidx.core.view.ViewCompat;
-import androidx.core.view.WindowInsetsCompat;
 
-public class MainActivity extends AppCompatActivity {
+public class MainActivity extends AppCompatActivity implements View.OnClickListener {
+
+    private EditText firstName;
+    private EditText lastName;
+    private EditText latitude;
+    private EditText longitude;
+    private EditText mail;
+    private EditText password;
+    private TextView labelFirstName;
+    private TextView labelLastName;
+    private TextView labelLatitude;
+    private TextView labelLongitude;
+    private TextView labelMail;
+    private TextView labelPassword;
+    private static final int PASSWORD_MIN_SIZE = 8;
+    private static final String REGEX_MAIL = "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$";
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         EdgeToEdge.enable(this);
-        setContentView(R.layout.activity_main);
-        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
-            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
-            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
-            return insets;
-        });
+        setContentView(R.layout.sign_up);
+        firstName = findViewById(R.id.first_name);
+        lastName = findViewById(R.id.last_name);
+        latitude = findViewById(R.id.latitude);
+        longitude = findViewById(R.id.longitude);
+        mail = findViewById(R.id.mail);
+        password = findViewById(R.id.password);
+
+        labelFirstName = findViewById(R.id.label_first_name);
+        labelLastName = findViewById(R.id.label_last_name);
+        labelLatitude = findViewById(R.id.label_position_lat);
+        labelLongitude = findViewById(R.id.label_position_long);
+        labelMail = findViewById(R.id.label_mail);
+        labelPassword = findViewById(R.id.label_password);
+
+        findViewById(R.id.sign_up_button).setOnClickListener(this);
+
+    }
+
+    /**
+     * Checks that the parameters entered are valid.
+     * Creates an account or notifies the user of input errors.
+     * @param v
+     */
+    public void onClick(View v) {
+        String errorMessage = "";
+        // reset the style of the text field
+        resetFieldStyle();
+
+        //TODO use the good error messages from string file
+        errorMessage += checkFirstName() + checkLastName() + checkLatitude()
```

Comment:
Dont' make a chain of concatenation, is less readable.

Can you make it like that ? 
```java
errorMessage += checkFirstName()
errorMessage += checkLastName()
errorMessage += checkLatitude()
//...
```

With the a `StringBuilder` it will look like that :
```java
errorMessage.append(checkFirstName())
errorMessage.append(checkLastName())
errorMessage.append(checkLatitude())
//...
```

Reply:
I change for a StringBuilder variable.